cub/benchmarks/bench/scan/applications/P1/s5-operator-associative-scan-segmented.cu

```diff
@@ -0,0 +1,236 @@
+    end_offset_it_t,
+    begin_offset_it_t,
+    op_t,
+    wrapped_init_t,
+    accum_t,
+    cub::ForceInclusive::Yes,
+    offset_t>;
+#endif
+
+  const auto elements       = static_cast<std::size_t>(state.get_int64("Elements{io}"));
+  const auto total_size     = elements * state_dim;
+  cudaStream_t bench_stream = state.get_cuda_stream().get_stream();
+
+  thrust::device_vector<value_t> input_A  = impl::generate_data<value_t>(total_size);
+  thrust::device_vector<value_t> input_Bu = impl::generate_data<value_t>(total_size);
+  thrust::device_vector<value_t> output_A(total_size, thrust::no_init);
+  thrust::device_vector<value_t> output_Bu(total_size, thrust::no_init);
+
+  const int nrows = elements;
+  const int ncols = state_dim;
+
+  auto col_major_iter =
+    cuda::make_transform_iterator(cuda::make_counting_iterator(0), impl::column_major_transform{nrows, ncols});
+
+  auto A_in_iter  = cuda::make_permutation_iterator(thrust::raw_pointer_cast(input_A.data()), col_major_iter);
+  auto Bu_in_iter = cuda::make_permutation_iterator(thrust::raw_pointer_cast(input_Bu.data()), col_major_iter);
+  auto input_iter = cuda::make_zip_iterator(A_in_iter, Bu_in_iter);
+
+  auto A_output_iter  = cuda::make_permutation_iterator(thrust::raw_pointer_cast(output_A.data()), col_major_iter);
+  auto Bu_output_iter = cuda::make_permutation_iterator(thrust::raw_pointer_cast(output_Bu.data()), col_major_iter);
+  auto output_iter    = cuda::make_zip_iterator(A_output_iter, Bu_output_iter);
+
+  thrust::device_vector<offset_t> begin_offsets(state_dim, thrust::no_init);
+  thrust::device_vector<offset_t> end_offsets(state_dim, thrust::no_init);
+
+  thrust::sequence(begin_offsets.begin(), begin_offsets.end(), 0, nrows);
+  thrust::sequence(end_offsets.begin(), end_offsets.end(), nrows, nrows);
+
+  state.add_element_count(elements);
+  state.add_global_memory_reads<value_t>(total_size, "A_in");
+  state.add_global_memory_reads<value_t>(total_size, "Bu_in");
+  state.add_global_memory_writes<value_t>(total_size, "A_out");
+  state.add_global_memory_writes<value_t>(total_size, "Bu_out");
+
+  size_t tmp_size;
+  dispatch_t::dispatch(
+    nullptr,
+    tmp_size,
+    input_iter,
+    output_iter,
+    state_dim,
+    thrust::raw_pointer_cast(begin_offsets.data()),
+    thrust::raw_pointer_cast(end_offsets.data()),
+    thrust::raw_pointer_cast(begin_offsets.data()),
+    op_t{},
+    wrapped_init_t{},
+    bench_stream);
+
+  thrust::device_vector<nvbench::uint8_t> tmp(tmp_size, thrust::no_init);
+  nvbench::uint8_t* d_tmp = thrust::raw_pointer_cast(tmp.data());
+
+  state.exec(nvbench::exec_tag::sync, [&](nvbench::launch& launch) {
+    dispatch_t::dispatch(
+      d_tmp,
+      tmp_size,
+      input_iter,
+      output_iter,
+      state_dim,
+      thrust::raw_pointer_cast(begin_offsets.data()),
+      thrust::raw_pointer_cast(end_offsets.data()),
+      thrust::raw_pointer_cast(begin_offsets.data()),
+      op_t{},
+      wrapped_init_t{},
```

Comment:
Nit: move these calls outside of `state.exec`. Unless compiler optimizes these away, they contribute to the benchmark execution wall-time.